symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/HttpClientExampleMain.java

```diff
@@ -0,0 +1,45 @@
+package com.symphony.bdk.examples;
+
+import static com.symphony.bdk.core.config.BdkConfigLoader.loadFromSymphonyDir;
+
+import com.symphony.bdk.core.SymphonyBdk;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.auth.jwt.JwtHelper;
+import com.symphony.bdk.core.config.exception.BdkConfigException;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.AuthenticateRequest;
+import com.symphony.bdk.gen.api.model.Token;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.util.GenericClass;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+
+public class HttpClientExampleMain {
+
+  public static void main(String[] args) throws BdkConfigException, IOException, GeneralSecurityException,
+      ApiException, AuthUnauthorizedException, AuthInitializationException {
+
+    JwtHelper jwtHelper = new JwtHelper();
+    BdkConfig config = loadFromSymphonyDir("config.yaml");
+    final SymphonyBdk bdk = new SymphonyBdk(config);
+    PrivateKey privateKey = jwtHelper.parseRsaPrivateKey(
+        IOUtils.toString(new FileInputStream(config.getBot().getPrivateKeyPath()), StandardCharsets.UTF_8));
+    final String jwt =
+        jwtHelper.createSignedJwt(config.getBot().getUsername(), JwtHelper.JWT_EXPIRATION_MILLIS, privateKey);
+    final AuthenticateRequest req = new AuthenticateRequest();
+    req.setToken(jwt);
+
+    Token token = bdk.http()
```

Comment:
if the http client is meant to be used for other usecases (outside of the SDK), perhaps we don't need to make it accessible in the BDK object (you would just create the Http client using the builder on its own)

Reply:
I only added the constructor for unittest purpose. Jacoco complained that the code coverage on `HttpClient` didn't reach 90% even when intellij said 100%, I think it's because the default constructor is not tested (not really sure about it)

Reply:
this idea is suggested by @symphony-thibault to make the bdk usage more fluent. @symphony-thibault WDYT ?

Reply:
This way developers don't need to pass the `ApiClientBuilderProvider`, it uses the one defined in `SymphonyBdk` found from ServiceLoader